src/quick-lint-js/fe/parse.cpp

```diff
@@ -334,6 +334,39 @@ void parser::error_on_sketchy_condition(expression* ast) {
   }
 }
 
+void parser::warn_on_pointless_string_condition(
+    expression::conditional* ast) {
+  auto is_or_operator = [](string8_view s) {
+    return s == u8"||"_sv;
+  };
+
+  for (span_size i = 0; i < ast->child_count() + 1; i++) {
+    expression* lhs = ast->child(i)->without_paren();
+    expression* rhs = ast->child(i + 1)->without_paren();
```

Comment:
Must fix: Iterating to `ast->child_count() + 1` looks wrong. It would mean we potentially access `ast->child(ast->child_count())` (lhs) and `ast->child(ast->child_count() + 1)` (rhs), which are out of bounds.

I think you meant `ast->child_count() - 1` instead.